### PR TITLE
{numlib} [{foss,intel}-2015a] CGAL 4.6 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/c/CGAL/CGAL-4.6-foss-2015a-Python-2.7.9.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+toolchainopts = {'strict': True}
+
+sources = [SOURCE_TAR_BZ2]
+source_urls = ['https://gforge.inria.fr/frs/download.php/file/34150/']
+
+pythonversion = '2.7.9'
+versionsuffix = "-Python-%s" % pythonversion
+
+dependencies = [
+    ('GMP', '6.0.0a'),
```
Comment (boegel) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.6-foss-2015a-Python-2.7.9.eb]: specify the `GCC` subtoolchain?

File: easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+toolchainopts = {'strict': True}
+
+sources = [SOURCE_TAR_BZ2]
+source_urls = ['https://gforge.inria.fr/frs/download.php/file/34150/']
+
+pythonversion = '2.7.9'
+versionsuffix = "-Python-%s" % pythonversion
+
+dependencies = [
+    ('GMP', '6.0.0a'),
```
Comment (boegel) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb]: specify the `GCC` subtoolchain?

File: easybuild/easyconfigs/g/GLib/GLib-2.44.0-foss-2015a.eb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+version = '2.44.0'
+
+homepage = 'http://www.gtk.org/'
+description = """GLib is one of the base libraries of the GTK+ project"""
+
+toolchain = {'name': 'foss', 'version': '2015a'}
+toolchainopts = {'optarch': True, 'pic': True}
+
+source_urls = ['http://ftp.gnome.org/pub/gnome/sources/glib/%(version_major_minor)s/']
+sources = ['glib-%(version)s.tar.xz']
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/GLib/GLib-2.44.0-foss-2015a.eb]: `SOURCELOWER_TAR_XZ` ?

File: easybuild/easyconfigs/m/MPFR/MPFR-3.1.2-intel-2015a-GMP-6.0.0a.eb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+source_urls = ['http://www.mpfr.org/mpfr-%(version)s/']
+sources = [SOURCELOWER_TAR_BZ2]
+
+patches = ['MPFR_ictce_remove-deprecated-mp.patch']
+
+gmp='GMP'
+gmpver='6.0.0a'
+versionsuffix='-%s-%s' % (gmp,gmpver)
+
+dependencies = [(gmp, gmpver, '', ('GCC', '4.9.2'))]
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MPFR/MPFR-3.1.2-intel-2015a-GMP-6.0.0a.eb]: maybe add a comment that GMP and icc are not friends and thus we're using GCC?

**Comment by wpoely86** [comment]
ok to test

**Comment by wpoely86** [comment]
lgtm

**Comment by hajgato** [comment]
polishing done

**Comment by wpoely86** [comment]
I've got a test running if it works, ready to merge

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 18 out of 22 (14 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/aedb8c5366618c8af77b for a full test report.

**Comment by boegel** [comment]
@wpoely86: you were saying? ;)

File: easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+dependencies = [
+#GMP does not compile correctly with ICC on Haswell architecture
+    ('GMP', '6.0.0a', '', ('GCC', '4.9.2')),
+    ('Boost', '1.58.0', versionsuffix),
+    ('MPFR', '3.1.2', '-GMP-6.0.0a'),
+    ('Qt', '4.8.6', '-GLib-2.44.0'),
+]
+
+builddeps = [
```
Comment (wpoely86) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb]: change to builddependencies everywhere!

@boegel unit test check for this?

Comment (boegel) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb]: to check what? typos? several easyconfig files use 'local variables'

if the typo is close enough (e.g. `buildependencies`), you'll get an error already:

```
ERROR: Failed to process easyconfig /tmp/bzip2-1.0.6.eb: You may have some typos in your easyconfig file: sources_urls -> source_urls
```

Comment (wpoely86) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb]: well, `builddeps` is quite close?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 23 out of 23 (14 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/4e1496d0ed367a97e6ea for a full test report.

**Comment by wpoely86** [comment]
@boegel ready for merge

File: easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb (hunk 5 above)
Comment (boegel) [inline easybuild/easyconfigs/c/CGAL/CGAL-4.6-intel-2015a-Python-2.7.9.eb]: so is `sources` and `source_urls` ;)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 14 out of 14 (14 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/784ca625021a68af3fa2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 14 out of 14 (14 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e2f4ebf7377328298cd0 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato and @wpoely86!

**Pull request merged**